c2rust-analyze/src/main.rs

```diff
@@ -321,6 +321,46 @@ where
     }
 }
 
+fn mark_foreign_fixed<'tcx>(
+    gacx: &mut GlobalAnalysisCtxt<'tcx>,
+    gasn: &mut GlobalAssignment,
+    tcx: TyCtxt<'tcx>,
+) {
+    // FIX the inputs and outputs of function declarations in extern blocks
+    for did in gacx.foreign_mentioned_tys.clone() {
+        if let DefKind::Fn = tcx.def_kind(did) {
```

Comment:
Does `foreign_mentioned_tys` really include `DefId`s of functions as well as types?  I thought it only stored the types (in which case we need some other way of visiting `extern`-block `fn`s here).  If `foreign_mentioned_tys` really does store both fns and types, then I think it should be renamed to make that clear.

Reply:
You're right; it only stores ADTs `DefId`s, not `fn` `DefId`s at all.  We could change this loop to iterating over all foreign items and filtering for just `DefKind::Fn` (and `DefKind::AssocFn`?) like we do in `fn foreign_mentioned_tys`.  Also, given this confusion, I think renaming the `tys` in `foreign_mentioned_tys` to `adts` might be helpful.

Reply:
You're right, this have been fixed in 7f8bddbe